custom-elements/custom-elements-registry/define.html

```diff
@@ -0,0 +1,230 @@
+<!DOCTYPE html>
+<title>Custom Elements: Element definition</title>
+<link rel="help" href="https://html.spec.whatwg.org/multipage/scripting.html#element-definition">
+<meta name="author" title="Koji Ishii" href="mailto:kojiishi@gmail.com">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<div id="log"></div>
+<script>
+'use strict';
+(function () {
+  var testable = false;
+  test(function () {
+    assert_true('customElements' in window, '"window.customElements" exists');
+    assert_true('define' in window.customElements, '"window.customElements.define" exists');
+    testable = true;
+  }, '"window.customElements.define" exists');
+  if (!testable)
+    return;
+
+  // 1. If IsConstructor(constructor) is false,
+  // then throw a TypeError and abort these steps.
+  test(function () {
+    assert_throws(new TypeError, function () {
+      window.customElements.define();
+    });
+  }, 'If no arguments, throw a TypeError');
+  test(function () {
+    assert_throws(new TypeError, function () {
+      window.customElements.define("test-define-one-arg");
+    });
+  }, 'If one argument, throw a TypeError');
+  [
+    [ "undefined", undefined ],
+    [ "null", null ],
+    [ "object", {} ],
+    [ "string", "string" ],
+    [ "arrow function", () => {} ], // IsConstructor returns false for arrow functions
+    [ "method", ({ m() { } }).m ], // IsConstructor returns false for methods
+  ].forEach(function (t) {
+    test(function () {
+      assert_throws(new TypeError, function () {
+        window.customElements.define("test-define-ctor-" + t[0], t[1]);
+      });
+    }, 'If constructor is ' + t[0] + ', throw a TypeError');
+  });
+
+  // 2. If name is not a valid custom element name,
+  // then throw a SyntaxError and abort these steps.
+  var validCustomElementNames = [
```

Comment:
Done for var => let.
I'm not really sure about cases v.s. underscores, testharness.js uses underscores, but JS/DOM doesn't, it always confuses me. I usually use underscores that is to extend testharness (e.g., assert_xxx) but camelCase for others, but I can't find guidelines.